WordPress/gutenberg
# [Try] HTTP based PHP signaling server for colaborative editing

This PR proposes a signaling server for WebRTC connections using PHP and HTTP. The work relies on the y-webrtc repository. Since certain required methods were not exported, we copied code from the y-webrtc package, adding a few exports within the y-webrtc folder.
This work employs WordPress Ajax requests to send messages to the server, and an EventSource to periodically retrieve messages. We have created a new class, HttpSignalingConn, to handle this type of signaling, along with a straightforward PHP signaling server. This server attempts to manage race conditions, ensuring message integrity.

Currently, we use temporary files to store messages on the server across distinct communication channels/clients. In the future, switching to a database might be logical, considering that in some WordPress setups, writing files might be unfeasible. However, for now, files offer greater flexibility and faster iteration.

cc: @youknowriad 

## Testing

- Enable the "Live collaboration and offline persistence" setting in "Gutenberg > Experiments".
- Open the site editor in two separate browsers and attempt collaboration. Observe real-time synchronization of changes.
- Build the plugin zip, then upload it to a WordPress test instance with HTTPS, like InstaWP.
- Confirm the possibility of collaboration between a computer a mobile device on a different network—e.g., mobile data on the phone and Wi-Fi on the computer. Occasionally, sync in such cases might lack smoothness or even fail; further enhancements are required.
